### ISSUE TITLE
Data race: globalObjectAPI used before initialized

### ISSUE THREAD

**Issue by klauspost** [minio/minio#8484]
[`initDailyLifecycle`/`startDailyLifecycle`](https://github.com/minio/minio/blob/26ed9e81b173521d9c32228329ad061a9a90b60f/cmd/daily-lifecycle-ops.go#L49) uses the `globalObjectAPI` before it is ready.

Even though the object is copied this is unsafe behavior.

## Expected Behavior

No data race.

## Current Behavior

```
λ go build -race && minio server data/disk1 data/disk2 data/disk3 data/disk4       
==================                                                                 
WARNING: DATA RACE                                                                 
Write at 0x000003b3c330 by main goroutine:                                         
  github.com/minio/minio/cmd.serverMain()                                          
      e:/gopath/src/github.com/minio/minio/cmd/server-main.go:376 +0xbba           
  github.com/minio/cli.HandleAction()                                              
      e:/gopath/pkg/mod/github.com/minio/cli@v1.22.0/app.go:500 +0x94              
  github.com/minio/cli.Command.Run()                                               
      e:/gopath/pkg/mod/github.com/minio/cli@v1.22.0/command.go:225 +0xd9e         
  github.com/minio/cli.(*App).Run()                                                
      e:/gopath/pkg/mod/github.com/minio/cli@v1.22.0/app.go:261 +0xa4b             
  github.com/minio/minio/cmd.Main()                                                
      e:/gopath/src/github.com/minio/minio/cmd/main.go:161 +0xaf                   
  main.main()                                                                      
      e:/gopath/src/github.com/minio/minio/main.go:37 +0x69                        
                                                                                   
Previous read at 0x000003b3c330 by goroutine 87:                                   
  github.com/minio/minio/cmd.startDailyLifecycle()                                 
      e:/gopath/src/github.com/minio/minio/cmd/daily-lifecycle-ops.go:59 +0x85     
                                                                                   
Goroutine 87 (running) created at:                                                 
  github.com/minio/minio/cmd.initDailyLifecycle()                                  
      e:/gopath/src/github.com/minio/minio/cmd/daily-lifecycle-ops.go:50 +0x49     
  github.com/minio/minio/cmd.serverMain()                                          
      e:/gopath/src/github.com/minio/minio/cmd/server-main.go:368 +0xb8d           
  github.com/minio/cli.HandleAction()                                              
      e:/gopath/pkg/mod/github.com/minio/cli@v1.22.0/app.go:500 +0x94              
  github.com/minio/cli.Command.Run()                                               
      e:/gopath/pkg/mod/github.com/minio/cli@v1.22.0/command.go:225 +0xd9e         
  github.com/minio/cli.(*App).Run()                                                
      e:/gopath/pkg/mod/github.com/minio/cli@v1.22.0/app.go:261 +0xa4b             
  github.com/minio/minio/cmd.Main()                                                
      e:/gopath/src/github.com/minio/minio/cmd/main.go:161 +0xaf                   
  main.main()                                                                      
      e:/gopath/src/github.com/minio/minio/main.go:37 +0x69                        
```

## Possible Solution

Protect use of `globalObjectAPI` or postpone starting the "DailyLifecycle" until it has been initialized.

## Steps to Reproduce (for bugs)

1. Start the server with race detection enabled.

## Regression

#7834 first added the problematic code. #8469 attempted a workaround, but didn't really fix the issue.

## Your Environment

λ git rev-parse master
26ed9e81b173521d9c32228329ad061a9a90b60f

**Comment by vadmeste** [comment]
Fixing it..

**Comment by harshavardhana** [comment]
This shall be fixed when #8478 is merged
